### PR TITLE
Lps 200318 Add tests for define api endpoint filters in api builder

### PR DESCRIPTION
**Pull request by yanan**
Background:
- Add tests to define prefilter

Test Plan
- background: Given objects definitons Student, Subject with field name created
And Given one-to-many relationship studentSubjects created
And Given Student entries 'aabc', 'abc', 'bbbabc','zzzz'  with externalReferenceCode 'aabc', 'abc', 'bbbabc', 'zzzz' created
And Given Subject entries 'abc', 'zzzz', 'bbbabc' created and related to students 'abc', 'bbbabc', 'zzzz'
Given ApiApplication with an endpoint of scope Company created and published
And Given a schema based on Student object definition with student name, subject name as schema properties student_name, subject_name
- When navigating to the endpoint Configuration tab
Filters box to Add a filter using OData is present
- When setting the Filters to 'contains(name,'abc')'
Then students 'aabc', 'abc', 'bbbabc' are listed in the endpoint response
- When setting the Filters to 'contains(studentSubjects/name,'abc')'
Then students 'abc', 'zzzz' are listed in the endpoint response
- And Given setting the Filters to 'contains(studentSubjects/name,'abc')'
And Given endpoint returning students 'abc', 'zzzz'
When I empty the Filter box in endpoint configuration
Then Student entries 'aabc', 'abc', 'bbbabc', 'zzzz' are listed in the endpoint response
- And Given setting the Filters to 'contains(studentSubjects/name,'abc') and name eq 'zzzz''
And Given endpoint returning student 'zzzz'
When I empty the Filters
And When I call endpoint with '?filter=contains(subject_name,'abc') and student_name eq 'zzzz'' in API Explorer
Then endpoint returns student 'zzzz'
- And Given setting the Filters to 'contains(name,'abc')'
When I call endpoint with '?filter=contains(subject_name,'z')' in API Explorer
Then endpoint returns student 'bbbabc'
- When setting the Filters to 'externalReferenceCode eq 'abc''
Then endpoint returns student 'abc'

Jira Ticket:
- https://liferay.atlassian.net/browse/LPS-200318

### CODE REVIEW

**Comment by yanan** [comment]
ci:test:headless-builder-functional:nocompile

File: modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/macros/APIApplicationEndpoint.macro [hunk 1]
```diff
@@ -19,6 +19,10 @@ definition {
 
 		var api = "${baseURL}${path}";
 
+		if (isSet(filter)) {
```
Comment (magjed4289) [inline modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/macros/APIApplicationEndpoint.macro:22]: 👍🏻

File: modules/apps/object/object-rest-test/src/testFunctional/macros/CustomObjectAPI.macro [hunk 2]
```diff
@@ -253,6 +253,27 @@ definition {
 		return ${response};
 	}
 
+	@summary = "Default summary"
```
Comment (magjed4289) [inline modules/apps/object/object-rest-test/src/testFunctional/macros/CustomObjectAPI.macro:256]: 👍🏻

File: modules/apps/object/object-rest-test/src/testFunctional/macros/CustomObjectAPI.macro [hunk 3]
```diff
@@ -253,6 +253,27 @@ definition {
...
+	macro createObjectRelatedToAnotherObject(en_US_plural_label = null, name = null, relatedObject = null, relatedObjectErc=null, relationship=null) {
+		Variables.assertDefined(parameterList = "${en_US_plural_label},${name},${relationship},${relatedObject},${relatedObjectErc}");
```
Comment (magjed4289) [inline modules/apps/object/object-rest-test/src/testFunctional/macros/CustomObjectAPI.macro:258]: ${relatedObject} is relatedObjectName? If so, please update

Comment (magjed4289) [reply modules/apps/object/object-rest-test/src/testFunctional/macros/CustomObjectAPI.macro:258]: ${relationship} is relationshipName? If so, please update

**Review by magjed4289 — CHANGES_REQUESTED** [review]
Just small changes requested, please see comments. Thank you!

**Comment by yanan** [comment]
ci:test:sf

File: modules/apps/object/object-rest-test/src/testFunctional/macros/CustomObjectAPI.macro (hunk 3 above)
Comment (yanan) [reply modules/apps/object/object-rest-test/src/testFunctional/macros/CustomObjectAPI.macro]: Updated and rebased, thanks!

**Comment by yanan** [comment]
ci:test:headless-builder-functional:nocompile

**Comment by yanan** [comment]
ci:test:sf

**Comment by yanan** [comment]
ci:test:headless-builder-functional:nocompile

**Review by magjed4289 — APPROVED** [review]

**Comment by magjed4289** [comment]
ci:forward

**Pull request closed without merging**